lib/js/src/manager/screen/_ScreenManagerBase.js

```diff
@@ -489,6 +498,107 @@ class _ScreenManagerBase extends _SubManagerBase {
         await this._subscribeButtonManager._removeButtonListener(buttonName, listener);
         return this;
     }
+
+    /**
+     * Present the alert on the screen.
+     * If the alert contains an audio indication with a file that needs to be uploaded,
+     * it will be uploaded before presenting the alert. If the alert contains soft buttons with images,
+     * they will be uploaded before presenting the alert. If the alert contains an icon, that will be uploaded before presenting the alert.
+     * The handler will be called when the alert either dismisses from the screen or it has failed to present.
+     * If the error value in the handler is present, then the alert failed to appear or was aborted,
+     * if not, then the alert dismissed without error. The error will contain `userInfo` with information on how long to wait before retrying.
+     * @param {AlertView} alert - the AlertView to present.
+     */
+    async presentAlert (alert) {
+        await this._alertManager.presentAlert(alert);
```

Comment:
The presentAlert method was updated to include an AlertCompletionListener and will return the success value through the listener instead of a Promise.